ckanext/datastore/cli.py

```diff
@@ -236,5 +218,33 @@ def upgrade():
         count, skipped, noinfo))
 
 
+@datastore.command(
+    'fts-index',
+    short_help='create or remove full-text search indexes after changing '
+    'the ckan.datastore.default_fts_index_field_types setting'
+)
+@click.option(
+    '--timeout', metavar='SECONDS',
+    type=click.FloatRange(0, 2147483.647),  # because postgres max int
+    default=_TIMEOUT / 1000, show_default=True,
+    help='maximum index creation time in seconds',
+)
+def fts_index(timeout: float):
+    '''Use to create or remove full-text search indexes after changing
+    the ckan.datastore.default_fts_index_field_types setting.
+    '''
+    site_user = logic.get_action('get_site_user')({'ignore_auth': True}, {})
+    resource_ids = get_all_resources_ids_in_datastore()
+
+    for i, resid in enumerate(get_all_resources_ids_in_datastore(), 1):
+        print(f'\r{resid} [{i}/{len(resource_ids)}] ...', end='')
+        logic.get_action('datastore_create')(
```

Comment:
We don't need to do anything now but it feels a bit uneasy to call `datastore_create` without params in order to trigger the indexes updating, as this will also do anything that a `datastore_create` call does. Maybe now that is just the indexes but who knows if something else is added in the future. A specific `datastore_update_indexes` call would be safer but again, we can improve this in the future.

Reply:
I was also uneasy with this, except that creating indexes is part of what `datastore_create` does automatically, so I don't know how you would move that functionality to a separate call without breaking the API.